### DISCUSSION TITLE
Simulating Backlash in Webots for an Existing Robot UR5e

### DISCUSSION THREAD

**Discussion by niklasconbotics** [cyberbotics/webots#6222] [General]
I'm currently working on a project involving the simulation of mechanical behavior in robotic systems using Webots, and I have a question concerning backlash.

I understand that Webots has a very detailed and robust simulation environment, which includes many physical properties of components. However, I'm trying to figure out if it is possible to simulate the effect of backlash on an existing robot in Webots. Backlash is the slack or play in a mechanical system, most commonly seen in gear trains and leadscrews, and it can have significant implications on the accuracy and precision of the robot's motion.

From my understanding, Webots may not directly support this feature in its physics engine. But, is there any workaround or trick that one could use to simulate this effect? For instance, adjusting certain parameters or implementing additional code?

Alternatively, does anyone know of any other library that includes backlash simulation in a more straightforward manner? I am open to looking at different tools if they would be more suitable for my use case.

Thank you in advance for your help and suggestions. Looking forward to a constructive discussion.

**Comment by omichel** [comment]
Webots features [hinge joints with backlash](https://www.cyberbotics.com/doc/guide/hinge-joint-with-backlash) as well as [hinge2 joints with backlash](https://www.cyberbotics.com/doc/guide/hinge-2-joint-with-backlash). However, they are more costly to simulate than standard joints, therefore they are not included by default in robot models. But it should be pretty straightforward to replace the hinge joints of any robot model, including the UR5e robots, with hinge joints with backlash. In order to do so, I would recommend you to create a local copy of the [UR5e proto](https://github.com/cyberbotics/webots/blob/master/projects/robots/universal_robots/protos/UR5e.proto), rename it to something like `UR5eWithBacklash.proto` and replace the joints with joints with backlash. Something similar was done in the [Darwin-OP](https://github.com/cyberbotics/webots/blob/master/projects/robots/robotis/darwin-op/protos/Darwin-op.proto) robot, although in this case, the [backlash](https://github.com/cyberbotics/webots/blob/master/projects/robots/robotis/darwin-op/protos/Darwin-op.proto#L35) was implemented as a boolean proto parameter.
